tests/testthat/test-object_length_linter.R

```diff
@@ -78,6 +78,21 @@ test_that("function shorthand is caught", {
 test_that("rlang name injection is handled", {
   linter <- object_length_linter(length = 10L)
 
-  expect_lint("tibble('{foo() |> bar() |> baz()}' := TRUE)", NULL, linter)
+  expect_no_lint("tibble('{foo() |> bar() |> baz()}' := TRUE)", linter)
   expect_lint("DT[, 'a_very_long_name' := FALSE]", "names should not be longer than 10 characters", linter)
 })
+
+test_that("literals in assign() and setGeneric() are checked", {
+  linter <- object_length_linter(length = 10L)
+  lint_msg <- rex::rex("Variable and function names should not be longer")
+
+  expect_no_lint("assign('good_name', 2, env)", linter)
+  expect_no_lint("assign('good_name', 'badBadBadBadName', env)", linter)
```

Comment:
What about keyword arguments?

Reply:
I like to use `assign(envir = ..., ...)` when I want to make the environment obvious.

Reply:
https://github.com/search?q=org%3Acran+lang%3AR+%2Fassign%5C%28%5Cs*envir%5Cs*%3D%2F&type=code

We even do it ourselves.

Reply:
I did see that, is why I checked everywhere and noticed we are unusual 😅

Reply:
We and 6 other CRAN packages ^^
Let's at least cover the named `envir=` case.

NB maybe we can come up with some general argument matching analysis. This problem comes up frequently enough it might be worth it I think.

Reply:
Done in e62ec672713640dedb02a63b91fd45dd2a0a21a3